File: R/wrap_fillsink.R
#' wrap_fillsink
#'
#' This will make the fillsinks funktion available to R from the libtotopotoolbox subdirectory
#' 
#' @param tiff Takes a Spatrast as input 
#'
#' @return An array
#' @export

wrap_fillsink <- function(dem,bc) {
    
    
    d <- get_grid_data(dem) # Extract input data
    output <- single(length(d$z)) #create output array
    result <- .C("wrap_fillsink",outputR=as.single(output),as.single(d$z),as.single(bc),as.integer(d$dims))$outputR
    
    return(result)
}

File: src/wrap_fillsink.c
#include <stddef.h>
#include <stdint.h>

#include "topotoolbox.h"

#include "topotoolboxr.h"

void wrap_fillsink(float *output, float *dem, int *bcR, int *dimsR){
   
   ptrdiff_t dims [2]= {dimsR[0], dimsR[1]};
   uint8_t bc = *bcR;
   fillsinks(output,dem,&bc,dims);
}
